src/ostree/ot-builtin-commit.c

```diff
@@ -817,12 +828,64 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
             goto out;
         }
 
-#ifndef OSTREE_DISABLE_GPGME
       if (opt_key_ids)
         {
+          /* Initialize crypto system */
+          if (!opt_sign_name)
+            opt_sign_name = "ed25519";
+
+          sign = ostree_sign_get_by_name (opt_sign_name, error);
+          if (sign == NULL)
+            goto out;
+
           char **iter;
 
           for (iter = opt_key_ids; iter && *iter; iter++)
+            {
+              const char *keyid = *iter;
+              gsize key_len = 0;
+              g_autofree guchar *key = NULL;
+              g_autoptr (GVariant) secret_key = NULL;
+
+              if (!g_strcmp0(ostree_sign_get_name (sign), "dummy"))
+                {
+                  secret_key = g_variant_new_string (keyid);
+                }
+#if defined(HAVE_LIBSODIUM)
+              else if (!g_strcmp0 (ostree_sign_get_name (sign), "ed25519"))
+                {
+                  key = g_malloc0 (crypto_sign_SECRETKEYBYTES);
+                  if (sodium_hex2bin (key, crypto_sign_SECRETKEYBYTES,
+                                 keyid, strlen (keyid),
+                                 NULL, &key_len, NULL) != 0)
+                    {
+                      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                                   "Invalid KEY '%s'", keyid);
+
+                      goto out;
+                    }
+
+                  secret_key = g_variant_new_fixed_array (G_VARIANT_TYPE_BYTE, key, key_len, sizeof(guchar));
+                }
+#endif
+              if (!ostree_sign_set_sk (sign, secret_key, error))
```

Comment:
secret_key can be NULL here is the parameter was not `dummy` and not `ed22519`

Reply:
yes, and should de handled by particular module implementation.